src/python_testing/TC_ECOINFO_2_1.py

```diff
@@ -15,20 +15,92 @@
 #    limitations under the License.
 #
 
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: examples/fabric-admin/scripts/fabric-sync-app.py
+#     app-args: --app-admin=${FABRIC_ADMIN_APP} --app-bridge=${FABRIC_BRIDGE_APP} --stdin-pipe=dut-fsa-stdin --discriminator=1234
+#     script-args: >
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --string-arg th_server_app_path:${ALL_CLUSTERS_APP} dut_fsa_stdin_pipe:dut-fsa-stdin
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     script-start-delay: 5
+#     factoryreset: true
+#     quiet: false
+# === END CI TEST ARGUMENTS ===
+
+import asyncio
+import logging
+import random
+import tempfile
+
 import chip.clusters as Clusters
 from chip.clusters.Types import NullValue
 from chip.interaction_model import Status
 from chip.tlv import uint
 from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, type_matches
 from mobly import asserts
+from TC_MCORE_FS_1_1 import AppServer
 
 
 class TC_ECOINFO_2_1(MatterBaseTest):
 
+    @async_test_body
+    async def setup_class(self):
+        super().setup_class()
+
+        self.th_server = None
+        self.storage = None
+
+        # Get the path to the TH_SERVER app from the user params.
+        if th_server_app := self.user_params.get("th_server_app_path"):
```

Comment:
You should only be doing this if you are in CI, otherwise the real test should NOT be creating TH_SERVER. That is not in the test plan.

Reply:
The intention here is that the `th_server_app_path` is going to be supplied on the CI only. However, I can add `if self.is_ci` guard here.

Reply:
>  because this is the same name (th_server_app_path) as the other test there is bound to be someone that accidentally uses this outside of CI with this current PR

Added `self.is_pics_sdk_ci_only` guard before running CI-related setup.